spring-batch-infrastructure/src/main/java/org/springframework/batch/item/file/SimpleBinaryBufferedReaderFactory.java

```diff
@@ -123,25 +123,24 @@ private boolean isEndOfLine(StringBuilder buffer, StringBuilder candidate, int n
 			}
 
 			char c = (char) next;
-			if (ending.charAt(0) == c || candidate.length() > 0) {
+			if (ending.charAt(0) == c || !candidate.isEmpty()) {
 				candidate.append(c);
 			}
-
-			if (candidate.length() == 0) {
+			else {
 				buffer.append(c);
 				return false;
 			}
 
-			boolean end = ending.equals(candidate.toString());
-			if (end) {
+			if (ending.contentEquals(candidate)) {
 				candidate.delete(0, candidate.length());
+				return true;
 			}
-			else if (candidate.length() >= ending.length()) {
+			if (!ending.startsWith(candidate.toString())) {
```

Comment:
The change in this line is the only "actual" change. The other changes are only intended to clarify the logic.